plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts

```diff
@@ -190,18 +220,25 @@ export function createFetchTemplateAction(options: {
       );
 
       for (const location of allEntriesInTemplate) {
-        const shouldCopyWithoutRender = nonTemplatedEntries.has(location);
+        let renderFilename = !nonTemplatedEntries.has(location);
+        let renderContents = renderFilename;
 
-        const outputPath = resolvePath(
-          outputDir,
-          shouldCopyWithoutRender
-            ? location
-            : templater.renderString(location, context),
-        );
+        let localOutputPath = location;
+        if (extension) {
```

Comment:
I think all of these conditionals and reassignments could be reorganized a bit to simplify what's going on here. For example splitting it all into whether extension or not is set.

I do wonder why we don't support setting both the extension and the two other options though? Not seeing what would prevent us from supporting that

Reply:
I gave the conditionals a try, not sure if that's what you had in mind.

Regarding the options, why would you have cookiecutterCompat on a template that you can't run with cookiecutter? For copyWithoutRender, please check https://github.com/cookiecutter/cookiecutter/issues/1588 for my lack of understanding what that's supposed to do to begin with. The current impl here isn't compatible with one of the two behaviours that cookiecutter has. I'd not want to spec out what that code should be doing, let alone with a template that won't run in cookiecutter.

Reply:
Yeah if it complicates things just for the sake of cookiecutter compatibility I'd say it's very much fine to keep the current behaviour. My impression of copyWithoutRender was that is was a more permanent thing that's similar to what you're solving here, but the other way around, and that we want to keep it around.